### DISCUSSION TITLE
Why do certain file types that should be ignored, continue to say I need to do something about them?

### DISCUSSION THREAD

**Discussion by Rod-in-NM** [community/community#23846] [New to GitHub]
<p>Yesterday I started a new Git repo, then copied source files from a TFVC repo into the Git repo. (I believe I got rid of all the TFVC related files.) I’m using Visual Studio 2019. After I created the local repo and pushed to remote, I realized that I’d forgotten to include the <code>.gitignore</code> file, so I went onto GitHub, found the relevant <code>.gitignore</code> for Visual Studio, and put it into the root of the repo. I thought that would be the end of it.</p>
<p>But I’m wrong. Every time I open the project with VS 2019, I’m presented with some files which Git sees ask having been modified, but not staged. For example, this is what I’m seeing now:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/f/5/f5aaa40ad00c4d2bd45844835610f45d1bc3570e.png" data-download-href="/uploads/short-url/z3gzJW63zxkSdjqpVQCrIaCnko6.png?dl=1" title="Screenshot 2021-09-29 142440" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/13875530/181079689-1e83d628-57d6-4a60-89ab-1b2a4388c7ee.png" alt="Screenshot 2021-09-29 142440" data-base62-sha1="z3gzJW63zxkSdjqpVQCrIaCnko6" width="690" height="241" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/f/5/f5aaa40ad00c4d2bd45844835610f45d1bc3570e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2021-09-29 142440</span><span class="informations">832×291 43.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I’ve previously performed a <code>git restore &lt;file&gt;</code> on files like the .suo file, but it still comes back. Files like .suo, .dll and .cache are all in the <code>.gitignore</code> file, so why aren’t they being ignored? Is the order in which I added the <code>.gitignore</code> file important? How do I fix this so that it will stop bothering me?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Rod-in-NM" data-post="1" data-topic="203674">
<div class="title">
<div class="quote-controls"></div>
 Rod-in-NM:</div>
<blockquote>
<p>Files like .suo, .dll and .cache are all in the <code>.gitignore</code> file, so why aren’t they being ignored?</p>
</blockquote>
</aside>
<p>The files in your screenshot are marked as <em>modified</em>, so they have already been committed to the repository. Files that are part of the repository are never ignored, you have to delete them from the index first.</p>
<p>You could use <a href="https://git-scm.com/docs/git-rm#Documentation/git-rm.txt---cached"><code>git rm --cached</code></a> on those files to delete them from the index but not the working tree. You might want to also clean them from the history (or start over, given that the repository is still very new).</p>

**Comment by Rod-in-NM** [comment]
<p>Thank you very much, <a class="mention" href="/airtower-luna">@airtower-luna</a>, for your explanation. It makes sense. I now understand it.</p>
<p>Like you said the repo is still new enough. I think the quickest way for me to get this resolves is to delete it from my machine, the clone the repo again.</p>
<p>Thanks again!</p>
